core/tokens.go

```diff
@@ -0,0 +1,91 @@
+package core
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/mr-tron/base58/base58"
+	"github.com/segmentio/ksuid"
+	"github.com/textileio/textile-go/crypto"
+	"github.com/textileio/textile-go/repo"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// CreateCafeToken creates a random developer access token, returns a base58 encoded version,
+// and stores a bcrypt hashed version for later comparison
+func (t *Textile) CreateCafeToken() (string, error) {
+	key, err := crypto.GenerateAESKey()
+	if err != nil {
+		return "", err
+	}
+
+	id := ksuid.New().String()
+	created := time.Now()
+	rawToken := key[:32]
+
+	safeToken, err := bcrypt.GenerateFromPassword(rawToken, bcrypt.DefaultCost)
+	if err != nil {
+		return "", err
+	}
+
+	err = t.datastore.CafeDevTokens().Add(
+		&repo.CafeDevToken{
+			Id:      id,
+			Token:   safeToken,
+			Created: created,
+		})
+	if err != nil {
+		return "", err
+	}
+
+	return id + "+" + base58.FastBase58Encoding(rawToken), nil
```

Comment:
what you're thinking w/ adding the id to the token?

Reply:
need it for db access on validation. as per our offline conversation, will update this to use random bytes instead of ksuid